sema4ai/vscode-client/src/views.ts

```diff
@@ -86,6 +87,39 @@ export async function openPackageTreeSelection(robot?: RobotEntry) {
     }
 }
 
+export async function openRunbookTreeSelection(robot?: RobotEntry) {
+    const selectedRobot = robot ?? getSelectedRobot();
+
+    if (!selectedRobot) {
+        return;
+    }
+
+    const agentYamlPath = path.join(selectedRobot.robot.directory, "agent-spec.yaml");
+    const agentYamlUri = vscode.Uri.file(agentYamlPath);
+
+    if (!(await uriExists(agentYamlUri))) {
+        return;
+    }
+
+    const yamlContent = await vscode.workspace.fs.readFile(agentYamlUri);
+    const yamlString = yamlContent.toString();
+
+    const parsedYaml = yaml.load(yamlString);
+    const runbookPath = parsedYaml["agent-package"]?.agents?.[0]?.runbook ?? null;
+
+    if (runbookPath) {
```

Comment:
If the runbook path isn't found a warning should be shown to the user saying so... The return of the action should be an `ActionResult` and `success` should be `False` in this case and the `message` should have the contents to show to the user on the failure case.

Reply:
Yes we could have all these messages but I wanted to avoid doing another set of validation just to open a file. The file comes with all the necessary things on creation and then the validation is done directly in VSCode when you change the file, so the yaml should technically be always valid. I guess it doesn't really hurt to put more information.

Reply:
Let's find a middle-ground so that we don't end-up duplication validations on every level.

Reply:
Still, it's possible that the user requests information in an invalid state. If that happens, we should warn him that we couldn't complete the action due to that.